diff -ruN a/nss/nss_files/files-XXX.c b/nss/nss_files/files-XXX.c
--- a/nss/nss_files/files-XXX.c	2022-07-29 22:03:09.000000000 +0000
+++ b/nss/nss_files/files-XXX.c	2022-11-03 07:47:53.034666951 +0000
@@ -73,8 +73,20 @@
 
   if (*stream == NULL)
     {
-      *stream = __nss_files_fopen (DATAFILE);
+      const char *file = DATAFILE;
 
+      #ifdef HAB_DATAFILE
+      // use the Habitat environment variable such as `HAB_ETC_PROTOCOLS`
+      char *path = secure_getenv (HAB_DATAFILE);
+
+      // if the environment variable is set, then read from the /hab/pkgs/* folder instead
+      if (path && path[0]) {
+        file = path;
+      }
+      #endif
+      
+      *stream = __nss_files_fopen (file);
+    
       if (*stream == NULL)
 	status = errno == EAGAIN ? NSS_STATUS_TRYAGAIN : NSS_STATUS_UNAVAIL;
     }
diff -ruN a/nss/nss_files/files-proto.c b/nss/nss_files/files-proto.c
--- a/nss/nss_files/files-proto.c	2022-07-29 22:03:09.000000000 +0000
+++ b/nss/nss_files/files-proto.c	2022-11-03 07:48:22.914491376 +0000
@@ -21,6 +21,7 @@
 
 #define ENTNAME		protoent
 #define DATABASE	"protocols"
+#define HAB_DATAFILE "HAB_ETC_PROTOCOLS"
 
 struct protoent_data {};
 
diff -ruN a/nss/nss_files/files-service.c b/nss/nss_files/files-service.c
--- a/nss/nss_files/files-service.c	2022-07-29 22:03:09.000000000 +0000
+++ b/nss/nss_files/files-service.c	2022-11-03 07:48:50.490329347 +0000
@@ -22,6 +22,7 @@
 
 #define ENTNAME		servent
 #define DATABASE	"services"
+#define HAB_DATAFILE "HAB_ETC_SERVICES"
 
 struct servent_data {};
 
