--- ./src/libbsd.pc.in.orig	2023-05-19 15:12:41.222474426 +0000
+++ ./src/libbsd.pc.in	2023-05-19 15:16:58.994376827 +0000
@@ -7,6 +7,6 @@
 Description: Utility functions from BSD systems
 Version: @VERSION@
 URL: https://libbsd.freedesktop.org/
-Libs: -L${libdir} -lbsd
-Libs.private: @LIBBSD_LIBS@
-Cflags: -I${includedir}
+Libs: -L${libdir} -lbsd
+Libs.private: -L@libmd_lib@ @LIBBSD_LIBS@
+Cflags: -I${includedir} -I@libmd_include@
--- ./test/Makefile.in.orig	2023-05-19 15:27:24.710143618 +0000
+++ ./test/Makefile.in	2023-05-19 15:28:15.954124644 +0000
@@ -635,7 +635,7 @@
 
 AM_TESTS_ENVIRONMENT = \
 	export CC="$(CC)"; \
-	export CPPFLAGS="$(HEADERS_CPPFLAGS)"; \
+	export CPPFLAGS="$(HEADERS_CPPFLAGS) -isystem @libmd_include@"; \
 	export top_srcdir="$(top_srcdir)"; \
 	$(nil)
 
