diff -ru a/lib/crypto/configure b/lib/crypto/configure
--- a/lib/crypto/configure	2023-01-30 12:35:07.586201427 +0000
+++ b/lib/crypto/configure	2023-01-30 12:38:15.823823071 +0000
@@ -5594,21 +5594,7 @@
 
 
 
-std_ssl_locations="\
-/usr/local \
-/usr/sfw \
-/usr \
-/opt/local \
-/usr/pkg \
-/usr/local/openssl \
-/usr/local/opt/openssl \
-/opt/homebrew/opt/openssl \
-/usr/lib/openssl \
-/usr/openssl \
-/usr/local/ssl \
-/usr/lib/ssl \
-/usr/ssl \
-/"
+std_ssl_locations=
 
 
 # Check whether --with-ssl was given.
@@ -6112,11 +6098,11 @@
                       if test "x$ac_cv_sizeof_void_p" = "x8"
 then :
 
-	                        std_win_ssl_locations="/mnt/c/OpenSSL-Win64 /c/OpenSSL-Win64 /mnt/c/opt/local64/pgm/OpenSSL /opt/local64/pgm/OpenSSL /cygdrive/c/OpenSSL-Win64"
+	                        std_win_ssl_locations=
 
 else $as_nop
 
-                                std_win_ssl_locations="/mnt/c/OpenSSL-Win32 /c/OpenSSL-Win32 /mnt/c/OpenSSL /c/OpenSSL /cygdrive/c/OpenSSL /opt/local/pgm/OpenSSL /opt/local32/pgm/OpenSSL /mnt/c/opt/local/pgm/OpenSSL /mnt/c/opt/local32/pgm/OpenSSL /cygdrive/c/OpenSSL-Win32"
+                                std_win_ssl_locations=
 
 fi
 
